internal/mock/mocks.go

```diff
@@ -101,40 +101,56 @@ func (t Repository) PushSignature(ctx context.Context, mediaType string, blob []
 	return ocispec.Descriptor{}, ocispec.Descriptor{}, nil
 }
 
+type PluginMock struct {
+	Metadata        proto.GetMetadataResponse
+	ExecuteResponse interface{}
+}
+
+func (p *PluginMock) GetMetadata(ctx context.Context, req *proto.GetMetadataRequest) (*proto.GetMetadataResponse, error) {
+	return &p.Metadata, nil
+}
+
+func (p *PluginMock) VerifySignature(ctx context.Context, req *proto.VerifySignatureRequest) (*proto.VerifySignatureResponse, error) {
+	if resp, ok := p.ExecuteResponse.(*proto.VerifySignatureResponse); ok {
+		return resp, nil
+	}
+	return nil, nil
+}
+
+func (p *PluginMock) DescribeKey(ctx context.Context, req *proto.DescribeKeyRequest) (*proto.DescribeKeyResponse, error) {
+	panic("not implemented") // TODO: Implement
+}
+
+func (p *PluginMock) GenerateSignature(ctx context.Context, req *proto.GenerateSignatureRequest) (*proto.GenerateSignatureResponse, error) {
+	panic("not implemented") // TODO: Implement
+}
+
+func (p *PluginMock) GenerateEnvelope(ctx context.Context, req *proto.GenerateEnvelopeRequest) (*proto.GenerateEnvelopeResponse, error) {
+	panic("not implemented") // TODO: Implement
+}
+
 type PluginManager struct {
-	PluginCapabilities          []plugin.Capability
+	PluginCapabilities          []proto.Capability
 	GetPluginError              error
 	PluginRunnerLoadError       error
 	PluginRunnerExecuteResponse interface{}
 	PluginRunnerExecuteError    error
 }
 
-type PluginRunner struct {
-	Response interface{}
-	Error    error
-}
-
-func (pr PluginRunner) Run(ctx context.Context, req plugin.Request) (interface{}, error) {
-	return pr.Response, pr.Error
-}
-
-func (pm PluginManager) Get(ctx context.Context, name string) (*manager.Plugin, error) {
-	return &manager.Plugin{
-		Metadata: plugin.Metadata{
+func (pm PluginManager) Get(ctx context.Context, name string) (plugin.Plugin, error) {
+	return &PluginMock{
+		Metadata: proto.GetMetadataResponse{
 			Name:                      "plugin-name",
 			Description:               "for mocking in unit tests",
 			Version:                   "1.0.0",
 			URL:                       ".",
 			SupportedContractVersions: []string{"1.0"},
 			Capabilities:              pm.PluginCapabilities,
 		},
-		Path: ".",
-		Err:  nil,
+		ExecuteResponse: pm.PluginRunnerExecuteResponse,
```

Comment:
Given the changes above, this should be:
```suggestion
		ExecuteResponse: pm.PluginRunnerExecuteResponse,
		ExecuteError:    pm.PluginRunnerExecuteError,
```

Reply:
Updated.